test/correctness/shuffle.cpp

```diff
@@ -0,0 +1,65 @@
+#include "Halide.h"
+#include <stdio.h>
+
+using namespace Halide;
+
+int main(int argc, char **argv) {
+    Target target = get_jit_target_from_environment();
+    if (target.has_feature(Target::Feature::Vulkan)) {
+        std::printf("[SKIP] Vulkan seems to be not working.\n");
+        return 0;
+    }
+
```

Comment:
I’d suggest adding a few much simpler test cases that just test known patterns with known indices. Make it trivial to find an error in the shuffle without having to diagnose the calculation of exp.

Reply:
Should this file be called `vector_shuffle.cpp`?

Reply:
The challenge of this test was writing some code that actually emits shuffles in the generated code. Make it too simple, and Halide will simplify it all away. I don't remember exactly why I came up with this, but the fact that this bug is in there for years makes me believe it wasn't trivial to hit that code path, due to simplifications and other lowering passes.

Reply:
> Should this file be called `vector_shuffle.cpp`?

Sure, I'll rename it!